src/app/mesh/qgsmeshlayerproperties.cpp

```diff
@@ -204,6 +211,22 @@ void QgsMeshLayerProperties::apply()
   mVector3DWidget->apply();
 #endif
 
+  QgsDebugMsg( QStringLiteral( "processing rendering tab" ) );
+  /*
+   * Rendering Tab
+   */
+  QgsMeshSimplifySettings simplifySettings;
+  simplifySettings.setEnabled( mSimplifyMeshGroupBox->isChecked() );
+  simplifySettings.setReductionFactor( mSimplifyReductionFactorSpinBox->value() );
+  simplifySettings.setMeshResolution( mSimplifyMeshResolutionSpinBox->value() );
+  bool needMeshUpdating = ( ( simplifySettings.isEnabled() != mMeshLayer->meshSimplificationSettings().isEnabled() ) ||
```

Comment:
implement `==` operator for QgsMeshSimplifySettings

Reply:
not sure, `QgsMeshSimplificationSettings` has a third parameters that is not concerned here : (`mMeshResolution`). Here the aim is to rebuild the simplified meshes because parameters related to this building this have changed. `mMeshResolution` is not needed for the mesh simplification, but only for the rendering to choose the best simplified mesh.

Reply:
and if the first two parameters are the same, is there any possiblitiy that mMeshResolution will differ?

Reply:
what I would fear is the bug in the future, since we may add 4th parameter in the settings and forgot to update it here...

Reply:
`mMeshResolution` could differ if only this parameter is changed in the properties dialog.

but you're right for the futur bug, why about a method in the `QgsMeshSimplificationSettings` names like `compareMeshSimplificationBuildingParameter()` ? Or other name, this is surely too long...

Reply:
you do not change mesh simplification settings that often. i would just recreate it even you change mMeshResolution

Reply:
I could take a lot of time to recreate simplification depending of the mesh, and in the main thread. My point of view is that it is pity to have freezing time for nothing and because a implementation detail...
But I don't have a strong point of view, and if you think it is important I do like you say.

Reply:
lets keep it like this then